F/missing py license (Python, Perl and Bash scripts were missing declarations)

bkil-syslogng opened the pull request:
Also enabled automatic checking in the future for the most important non-C kinds (e.g., Java).

bazsi commented on modules/python/pylib/syslogng/__init__.py:
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+#############################################################################
```
I don't think we need a shebang line here and neither of other modules in the same package.

bkil-syslogng replied:
@bazsi 
1. Some of the Python files contained this, some of them did not. I wanted to make it uniform.
2. I recommend using a shebang in each and every script to make file magic consistent on all platforms.

In light of these, do you agree?

bazsi replied:
executable scripts contained this, whereas modules to be imported didn't.
having a shebang line indicates that the file would be executable, which
they are not.

-- 
Bazsi

On Mon, Mar 21, 2016 at 10:22 AM, Tamas Nagy <notifications@github.com>
wrote:

> In modules/python/pylib/syslogng/__init__.py
> <https://github.com/balabit/syslog-ng/pull/985#discussion_r56795686>:
>
> > @@ -0,0 +1,22 @@
> > +#!/usr/bin/env python
> > +#############################################################################
>
> @bazsi <https://github.com/bazsi>
> 1. Some of the Python files contained this, some of them did not. I wanted
> to make it uniform.
> 2. I recommend using a shebang in each and every script to make file magic
> consistent on all platforms.
>
> In light of these, do you agree?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/985/files/4bd9d6e8f9bcba316d6ae8b02b701ffa4d85f6ea#r56795686>
>

bazsi commented on tests/functional/control.py:
```diff
@@ -1,3 +1,25 @@
+#!/usr/bin/env python
+#############################################################################
```
no shebang needed either.

bkil-syslogng replied:
@bazsi And let me also add a nitpick of my own. I would generally recommend setting the executable bit on all of the scripts which could be meaningfully executed from the command line.

bazsi commented:
Apart from my nitpicks wrt shebang lines, this looks great, thank you.

bkil-syslogng commented:
@bazsi I've added two new commits to remove redundant shebangs. Please notify me after you have reviewed them, so I can squash the fixup.

bazsi commented:
ok on my part, thanks for the fixups.

lbudai commented on tests/copyright/policy:
```diff
@@ -22,6 +22,6 @@ modules/native
 tests
 modules/java/(tools|[^/]*$)
-modules/java-modules/(dummy|elastic|hdfs|http|kafka|[^/]*$)
+modules/java-modules/(dummy|elastic|elastic-v2|hdfs|http|kafka|[^/]*$)
```
Hmm... we should add every new module to this file?
That will always be a problem...

bkil-syslogng replied:
Yes, we do need to add each one when using the current format. This is because we have both GPL and non-GPL modules in the same folder, and I don't want to add overlapping patterns and row based prioritization, because that can bite us very quickly without ever noticing. As a side note, currently, overlapping declarations produce an error (e.g., the containing folder under GPL and subfolders under LGPL).

After these improvements are merged, this check (and Travis) should always tell you if you forgot to add a module and it should also point out which one. The omission was on my part that I forgot to check `.java` files, so that's why you didn't get a notification about it before.

lbudai commented:
@bkil-syslogng : could you squash the patches?

bkil-syslogng commented:
Squashed and rebased to master. Backup saved:
https://github.com/bkil-syslogng/syslog-ng/compare/master...bkil-syslogng:t/missing-py-license-bk0?expand=1
